### ISSUE TITLE
Missing support for java modules.

### ISSUE THREAD

**Issue by anders-hig-jackson** [mohamed-taman/Agile-Software-Dev-Refactoring#1]
The only thing I miss in this tutorial is support for java module, as you are not using JDK 8.

I uses Eclipse (and Emacs), so I don't know best way to support modules in other IDE:s.  In Eclipse you create a project for each module, giving the project same name as the module.  After that it's easy to import/export from/to that module from other modules.

**Comment by mohamed-taman** [comment]
Hi @anders-hig-jackson, every IDE allows you to create projects with modules. I have done that with many IDEs like the eclipse, as you have mentioned, IntelliJ IDEA, NetBeans, and VSCode.

Here is an example from Inlillij Idea https://www.jetbrains.com/help/idea/creating-and-managing-modules.html

But my advice here when developing a multi-module project is to use **maven and sub maven projects as modules**, so it will work on all IDEs by just importing the project into many IDEs without issues.

_I hope this will help._

**Comment by anders-hig-jackson** [comment]
Maven are not in any way a contradiction to Java modules.  The solve different problems.

Maven is about distribution of libraries.  Java modules are about restrict access to identifiers from different parts of a program and libraries.  And it also solves the Classpath hell of Java, which Maven doesn't address.

**Comment by mohamed-taman** [comment]
@anders-hig-jackson, okay, you are talking about Java Platform Module System **JPMS**, I missed it. From my experience, it is the same in all IDEs. I wrote article series about the subject and access restrictions, and here they are hopeful they help (I used IntelliJ here): 

[Java theory and practice: Java 9+ modularity](https://developer.ibm.com/components/java-platform/series/java-modularity/) check it out.
